src/mantine-core/src/components/NumberInput/NumberInput.tsx

```diff
@@ -345,6 +345,45 @@ export const NumberInput = forwardRef<HTMLInputElement, NumberInputProps>(
       }
     };
 
+    const getMobileOperatingSystem = () => {
+      const userAgent = navigator.userAgent || navigator.vendor;
+      if (/android/i.test(userAgent)) {
+        return 'Android';
+      }
+      if (/iPad|iPhone|iPod/.test(userAgent)) {
+        return 'iOS';
+      }
+      return 'unknown';
+    };
+
+    const handleInputMode = ():
```

Comment:
Move this function to a separate file and rename it to `getInputMode`

Reply:
Got it!